core-common/pom.xml

```diff
@@ -564,6 +528,21 @@
                                     </resources>
                                 </configuration>
                             </execution>
+                            <execution>
+                                <id>copy-jdk9-classes-to-meta-inf</id>
+                                <phase>prepare-package</phase>
+                                <goals>
+                                    <goal>copy-resources</goal>
```

Comment:
Are you sure that sources jar contains these jdk "overrides"?
Did you consider using buildhelper's add-resources goal together with its targetfolder configuration option?

Reply:
Affirmative. Sources are copied by copy-jdkX-sources targets.